devices/power.py

```diff
@@ -50,7 +51,18 @@ def get_power_device(ip_address, username=None, password=None, outlet=None):
     except UnicodeDecodeError as e:
         data = urlopen("http://" + ip_address).read()
     except HTTPError as e:
-        data = e.read().decode()
+        """check firefox installed on platform"""
+        detectFirefox(dirpath='/usr/bin')
+        """ Firefox """
+        options = webdriver.FirefoxOptions()
+        options.set_headless()
+        options.add_argument('--disable-gpu')
+        driver = webdriver.Firefox(firefox_options=options)
+        driver.get("http://" + username + ':' + password + '@' + ip_address)
+        data = driver.page_source
+        driver.find_element_by_id('memLogout').click()
+        time.sleep(1)
+        driver.quit()
```

Comment:
No, this is not what I meant.

If there were not match just login with the previous method you were using and check the output after logging in. DO NOT START FIREFOX FOR THIS.

Reply:
Will remove it.